src/bsoncxx/document/value.hpp

```diff
@@ -88,6 +107,14 @@ class BSONCXX_API value {
     ///
     BSONCXX_INLINE operator document::view() const noexcept;
 
+    // Function to help with serialization
+    template <typename T>
+    T get() {
+        T temp_object{};
```

Comment:
So this is awesome. It works just as intended. If you have time you could also allow partial construction of an object. In other words, you could have this method take in an optional value of type T and pass that to `from_bson`. So this method would become:
```
    template <typename T>
    T get(T o={}) {
        from_bson(o, this->view());
        return o;
    }
```
and allow the user to do this:
```
test::Person x;
test::Person y = expected_doc.get(x);
```
Note that the brackets (`<>`) aren't needed in this example since we can deduce the type T based on its argument. Feel free to hold off on this though. If you do decide to implement partial construction too you'd probably want to add a test like the code snippet.

Reply:
Is this a normal pattern and/or use case? It seems like doing this will result in both `Person x` and `Person y` having the same deserialized values from the `expected_doc`. It's useful for if the user would like to have 2 Persons with the same value, but I'm just not used to seeing this for assignment.

Reply:
OH, that makes sense. Thank you!